guides/user/canary/best-practices/index.md

```diff
@@ -16,6 +16,118 @@ critical metric fails, but the rest pass, the group gets a passing score overall
 You can put a critical metric in a group of only one to ensure that if it fails,
 the whole group fails every time.
 
+## Compare canary against baseline, not against production
+
+You might be tempted to compare the canary deployment against your current
+production deployment. Instead always compare the canary against an equivalent
+baseline, deployed at the same time.
+
+The baseline uses the same version and configuration that is currently running
+in production, but is otherwise identical to the canary:
+
+* Same time of deployment
+* Same size of deployment
+* Same type and amount of traffic
+
+In this way, you control for version and configuration only, and you reduce
+factors that could affect the analysis, like the cache warmup time, the heap
+size, and so on.
+
+## Run the canary for enough time
+
+You need at least 50 pieces of time series data per metric for the statistical
```

Comment:
Johannes' comment from our doc:

> Say, for the sake of argument, that I am sampling every millisecond. Would not 50 datapoints be pretty insigificant in that case? During that time, the app may have only dispatched a single request -- or maybe even none. Maybe memory usage doubles every second -- 50 datapoints will not catch that.
I'd expect that you need at least maybe 50 datapoints AND during that time the app should have done some serious work AND you should run the analysis for long enough to give you some clue about stability.

Reply:
I think that Johannes has a point here. And that it actually depends on the monitoring system.
Unless I'm mistaken, Stackdriver won't give you more than one data point per minute in the answer, which gives you a 50m minimum for a canary run. But this is not the case for Prometheus or Datadog AFAIK, where you can have more data points. It's not uncommon for Prometheus users to have a data point every 15s.
@duftler how do we deal with that?

Reply:
@csanden WDYT?

Reply:
@csanden is definitely the authority here. My point in the earlier thread was that it wouldn't affect the classification. But yes, you will be constrained by the precision of your tsdb. The classification logic should have a minimum of 50 data points for you to have confidence in the result, but the result will only be meaningful for a given use case if the baseline and canary are exposed to true production workload during the run. I agree with Johannes's point and maybe Chris can recommend some good wording here?